overlord/devicestate/devicestate_install_api_test.go

```diff
@@ -211,23 +234,38 @@ func (s *deviceMgrInstallAPISuite) testInstallFinishStep(c *C, opts finishStepOp
 		writeContentCalls++
 		if opts.encrypted {
 			c.Check(encSetupData, NotNil)
+
+			// Make sure we "observe" grub from boot partition
+			mockRunBootStruct := &gadget.LaidOutStructure{
+				VolumeStructure: &gadget.VolumeStructure{
+					Role: gadget.SystemBoot,
+				},
+			}
+			writeChange := &gadget.ContentChange{
+				// file that contains the data of the installed file
+				After: filepath.Join(dirs.RunDir, "mnt/ubuntu-boot/EFI/boot/grubx64.efi"),
+				// there is no original file in place
+				Before: "",
+			}
+			action, err := observer.Observe(gadget.ContentWrite, mockRunBootStruct,
+				filepath.Join(dirs.RunDir, "mnt/ubuntu-boot/"),
+				"EFI/boot/grubx64.efi", writeChange)
+			c.Check(err, IsNil)
+			c.Check(action, Equals, gadget.ChangeApply)
 		} else {
 			c.Check(encSetupData, IsNil)
 		}
 		return nil, nil
 	})
 	s.AddCleanup(restore)
 
-	// Note that ESP must be mounted in the same place as a seed partition
-	// so MarkRecoveryCapableSystem is happy when searching for a bootloader.
-	// TODO Should this be changed?
-	espDir := filepath.Join(dirs.RunDir, "mnt/ubuntu-seed")
+	seedNullDir := filepath.Join(dirs.RunDir, "mnt/ubuntu-seed")
```

Comment:
this probably can be just called seedDir as it doesn't matter here if there's a seed there or not?